src/main/python/run_es_regression.py

```diff
@@ -0,0 +1,235 @@
+
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Program for running Elasticsearch regressions.')
+    parser.add_argument('--ping', action='store_true', default=False, help='ping ES and exit')
+    parser.add_argument('--check-index-exists', default='', type=str, metavar='collection', help='check if index exists')
+    parser.add_argument('--delete-index', default='', type=str, metavar='collection', help='deletes index')
+    parser.add_argument('--create-index', default='', type=str, metavar='collection', help='creates index')
+    parser.add_argument('--insert-docs', default='', type=str, metavar='collection', help='insert documents into index')
+    parser.add_argument('--input', default='', type=str, metavar='directory', help='location of documents to insert into index')
+    parser.add_argument('--evaluate', default='', type=str, metavar='collection', help='search and evaluate on collection')
+    parser.add_argument('--regression', default='', type=str, metavar='collection', help='run end-to-end regression')
+
+    args = parser.parse_args()
+    es = ElasticsearchClient()
+
+    if args.ping:
+        logger.info('Pinging Elasticsearch instance...')
+        if es.is_alive():
+            logger.info('... appears to alive! :)')
+        else:
+            logger.info('... appears to dead! :(')
+    elif args.check_index_exists:
+        logger.info('Checking if index {} exists...'.format(args.check_index_exists))
+        if es.does_index_exist(args.check_index_exists):
+            logger.info('... yes indeed!')
+        else:
+            logger.info('... appears not.')
+    elif args.delete_index:
+        if es.delete_index(args.delete_index):
+            logger.info('... successful!')
+        else:
+            logger.info('... failed!')
+    elif args.create_index:
+        if es.create_index(args.create_index):
+            logger.info('... successful!')
+        else:
+            logger.info('... failed!')
+    elif args.insert_docs:
+        if not args.input:
+            raise Exception('Location of corpus not specified (use --input)!')
+        else:
+            es.insert_docs(args.insert_docs, args.input)
+    elif args.evaluate:
+        es.evaluate(args.evaluate)
+    elif args.regression:
+        logger.info('Running BM25 regression on {}...'.format(args.regression))
+        if not args.input:
+            raise Exception('Location of corpus not specified (use --input)!')
+        if not es.is_alive():
+            raise Exception('Elasticsearch does not appear to be alive!')
+        if es.does_index_exist(args.regression):
+            logger.info('Index {} already exists: deleting and recreating.'.format(args.regression))
+            es.delete_index(args.regression)
+        es.create_index(args.regression)
+        es.insert_docs(args.regression, args.input)
+        logger.info('Document ingestion complete. Sleeping now for 60s...')
+        time.sleep(60)
```

Comment:
question.. why do we want to pause here?

Reply:
This is to wait for ES to "catch up" - ingested documents are not available to search immediately. @r-clancy tried tweaking various refresh periods, etc., but the safest way is to just wait. I'll add a comment to document.

Reply:
I see, thanks for the explanation!